src/components/widgets/Form.js

```diff
@@ -0,0 +1,315 @@
+import React, { Component } from 'react'
+import { RadioGroup, Radio } from 'react-radio-group'
+import Button from '../Button.js'
+import "../../styles/form.css"
+
+export default class Form extends Component {
+  constructor( props ) {
+    super( props )
+    this.state = {}
+    this.setFieldState = this.setFieldState.bind( this )
+  }
+
+  componentWillMount() {
+    this.setupState()
+  }
+
+  setupState() {
+    const { sections } = this.props
+    sections.forEach( section => {
+      section.fields.forEach ( field => {
+        const { name, defaultValue } = field
+        this.setState({[ name ]: defaultValue })
+        if ( field.type === "nested-dropdown" ) {
+          const nestedDropdownValue = field.options.filter( option => {
+            return option.value === defaultValue
+          })[0].field
+          this.setState({
+            [ nestedDropdownValue.name ]: nestedDropdownValue.defaultValue
+          })
+        }
+      })
+    })
+  }
+
+  setFieldState( state, stateValue ) {
+    this.setState({
+      [ state ]: stateValue
+    })
+  }
+
+  submitWidget( state ) {
+    const widgetData = {}
+    for ( let key in state ) {
+      widgetData[key] = state[key]
+      widgetData.y = '10px'
+      widgetData.x = '20px'
+    }
+    console.log('You submitted a widget with this data:', widgetData)
+  }
+
+  render() {
+    const { type, sections, description, image } = this.props
+
+    return (
+      <div className="widget-form-container">
+        <div className="widget-form-header">
+          <h1>{type}</h1>
+        </div>
+        <form
+          onSubmit={() => this.submitWidget( this.state )}
+          className="widget-form">
+          {sections.map(( section, id ) => (
+              <Section
+                {...section}
+                setFieldState={this.setFieldState}
+                state={this.state}
+                key={`section-${id}`}
+              />
+            )
+          )}
+          <div className="widget-form-buttons">
+            <Button content="Add Widget" type="submit"/>
+            <Button content="Cancel" type="cancel"/>
+          </div>
+        </form>
+        <div className="widget-form-preview">
+          <Preview image={image} description={description}/>
+        </div>
+      </div>
+    )
+  }
+}
+
+class Section extends Component {
+  render() {
+    const { type, fields, setFieldState, state } = this.props
+
+    return (
+      <div className="widget-form-section">
+        <div className="widget-form-section-type">
+          {type === 'Setup'
```

Comment:
My intent is to avoid magic strings/numbers:
```jsx
// In an appropriate module - constants.js maybe?
const SETUP = 'setup'
const SETUP_HEADING = 'Setup - Now set up your widget'
const SOURCE_HEADING = 'Source - Where is your data coming from?'

export default { /* all those constants */ }
```
```jsx
import CONSTANTS from './constants/'

{type === SETUP ? SETUP_HEADING : SOURCE_HEADING }
```